cmd/influxd/inspect/dump_tsi/dump_tsi.go

```diff
@@ -0,0 +1,513 @@
+or it should specify one or more .tsi/.tsl files. If no flags are specified
+then summary stats are provided for each file.
+`,
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Parse filters.
+			if measurementFilter != "" {
+				re, err := regexp.Compile(measurementFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex: %w", err)
+				}
+				arguments.measurementFilter = re
+			}
+			if tagKeyFilter != "" {
+				re, err := regexp.Compile(tagKeyFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex: %w", err)
+				}
+				arguments.tagKeyFilter = re
+			}
+			if tagValueFilter != "" {
+				re, err := regexp.Compile(tagValueFilter)
+				if err != nil {
+					return fmt.Errorf("failed to parse regex: %w", err)
+				}
+				arguments.tagValueFilter = re
+			}
+
+			// Validate series file path.
+			if arguments.seriesFilePath == "" {
+				return fmt.Errorf("series file path required")
+			}
+
+			arguments.paths = args
+			if len(arguments.paths) == 0 {
+				return fmt.Errorf("at least one path required")
+			}
+
+			// Some flags imply other flags.
+			if arguments.showTagValueSeries {
+				arguments.showTagValues = true
+			}
+			if arguments.showTagValues {
+				arguments.showTagKeys = true
+			}
+			if arguments.showTagKeys {
+				arguments.showMeasurements = true
+			}
+
+			w := cmd.OutOrStdout()
+			arguments.w = &w
+			return arguments.run()
+		},
+	}
+
+	cmd.Flags().StringVar(&arguments.seriesFilePath, "series-file", "",
+		"Path to series file")
+	cmd.Flags().BoolVar(&arguments.showSeries, "series", false,
+		"Show raw series data")
+	cmd.Flags().BoolVar(&arguments.showMeasurements, "measurements", false,
+		"Show raw measurement data")
+	cmd.Flags().BoolVar(&arguments.showTagKeys, "tag-keys", false,
+		"Show raw tag key data")
+	cmd.Flags().BoolVar(&arguments.showTagValues, "tag-values", false,
+		"Show raw tag value data")
+	cmd.Flags().BoolVar(&arguments.showTagValueSeries, "tag-value-series", false,
+		"Show raw series data for each value")
+	cmd.Flags().StringVar(&measurementFilter, "measurement-filter", "",
+		"Regex measurement filter")
+	cmd.Flags().StringVar(&tagKeyFilter, "tag-key-filter", "",
+		"Regex tag key filter")
+	cmd.Flags().StringVar(&tagValueFilter, "tag-value-filter", "",
+		"Regex tag value filter")
+
+	return cmd
+}
+
+func (a *args) run() error {
+	sfile := tsdb.NewSeriesFile(a.seriesFilePath)
+	//sfile.Logger = logger.New(os.Stderr)
+	if err := sfile.Open(); err != nil {
+		return err
+	}
+	defer sfile.Close()
+
+	// Build a file set from the paths on the command line.
+	idx, fs, err := a.readFileSet(sfile)
+	if err != nil {
+		return err
+	}
+
+	if a.showSeries {
+		if err := a.printSeries(sfile); err != nil {
+			return err
+		}
+	}
+
+	// If this is an ad-hoc fileset then process it and close afterward.
+	if fs != nil {
+		defer fs.Release()
+		defer fs.Close()
```

Comment:
I think these `defer`s should happen before the `printSeries` call. Otherwise if `printSeries` exits with an error, the file set won't get cleaned up properly